lib/spack/docs/build_systems/cudapackage.rst

```diff
@@ -9,35 +9,120 @@
 CudaPackage
 -----------
 
-Different from other packages, ``CudaPackage`` does not represent a build
-system. Instead its goal is to simplify and unify usage of ``CUDA`` in other
-packages.
+Different from other packages, ``CudaPackage`` does not represent a build system.
+Instead its goal is to simplify and unify usage of ``CUDA`` in other packages.
 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Provided variants and dependencies
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can find source for the package at
+`<https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cuda.py>`__.
 
-``CudaPackage`` provides ``cuda`` variant (default to ``off``) to enable/disable
-``CUDA``, and ``cuda_arch`` variant to optionally specify the architecture.
-It also declares dependencies on the ``CUDA`` package ``depends_on('cuda@...')``
-based on the architecture as well as specifies conflicts for certain compiler versions.
+^^^^^^^^
+Variants
+^^^^^^^^
+
+This package provides the following variants:
+
+* **cuda**
+
+  This variant is used to enable/disable building with ``CUDA``. The default
+  is disabled (or ``False``).
+
+* **cuda_arch**
+
+  This variant supports the optional specification of the architecture.
+  Valid values are maintained in the ``cuda_arch_values`` property and
+  are the numeric character equivalent of the compute capability version
+  (e.g., '10' for version 1.0). Each provided value affects associated
+  ``CUDA`` dependencies and compiler conflicts.
+
+  GPUs and their compute capability versions are listed at
+  https://en.wikipedia.org/wiki/CUDA#GPUs_supported.
+
+^^^^^^^^^
+Conflicts
+^^^^^^^^^
+
+Conflicts are used to prevent builds with known bugs or issues. While
+base ``CUDA`` conflicts have been included with this package, you may
+want to add more for your software.
+
+For example, if your package requires ``cuda_arch`` to be specified when
+``cuda`` is enabled, you can add the following conflict to your package
+to terminate such build attempts with a suitable message:
+
+.. code-block:: python
+
+    conflicts('cuda_arch=none', when='+cuda',
+              msg='CUDA architecture is required')
+
+Similarly, if your software does not support all versions of the property,
+you could add ``conflicts`` to your package for those versions.  For example,
+suppose your software does not work with CUDA compute capability versions
+prior to `5.0`.  You can add the following code to display a custom message
+should a user attempt such a build:
+
+.. code-block:: python
```

Comment:
This is a great example and I think we could even implement a helper method in a follow-up to pass a minimum numeric SM value, reducing boiler-plate in packages.

Reply:
I think we can do that using metaclasses and requesting packages to just provide a map:
```
min_cuda_arch = [
    (min_value, when_spec_str),
    (min_value, when_spec_str),
    ...
]
```
Would that be a good enough interface?

Reply:
absolutely, great idea!

Reply:
Resolving this comment under the assumption the work discussed will be addressed in a follow-on PR for the metaclass and map approach suggested by alalazo.